gui-wm/hyprland/hyprland-0.26.0.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit meson toolchain-funcs
+
+DESCRIPTION="A dynamic tiling Wayland compositor that doesn't sacrifice on its looks"
+HOMEPAGE="https://github.com/hyprwm/Hyprland/releases"
+
+SRC_URI="https://github.com/hyprwm/${PN^}/releases/download/v${PV}/source-v${PV}.tar.gz -> ${PF}.gh.tar.gz"
+S="${WORKDIR}/${PN}-source"
+
+KEYWORDS="~amd64"
+LICENSE="BSD"
+SLOT="0"
+IUSE="X legacy-renderer systemd video_cards_nvidia"
+
+RDEPEND="
+	app-misc/jq
+	dev-libs/libevdev
+	dev-libs/libinput
+	dev-libs/wayland
+	dev-libs/wayland-protocols
+	dev-util/glslang
+	dev-util/vulkan-headers
+	gui-libs/gtk-layer-shell
+	media-libs/libdisplay-info
+	media-libs/libglvnd[X?]
+	media-libs/mesa[gles2,wayland,X?]
+	media-libs/vulkan-loader
+	x11-base/xcb-proto
+	x11-libs/cairo
+	x11-libs/libdrm
+	x11-libs/libxkbcommon
+	x11-libs/pixman
+	x11-misc/xkeyboard-config
+	virtual/libudev
+	X? (
+	   gui-libs/wlroots[x11-backend]
+	   x11-base/xwayland
+	   x11-libs/libxcb
+	   x11-libs/xcb-util-image
+	   x11-libs/xcb-util-renderutil
+	   x11-libs/xcb-util-wm
+	)
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-libs/hyprland-protocols
+	dev-libs/libliftoff
+	>=dev-libs/wayland-1.22.0
+	dev-vcs/git
+	>=gui-libs/wlroots-0.16.0[X?]
+"
+
+src_prepare() {
+	STDLIBVER=$(echo '#include <string>' | $(tc-getCXX) -x c++ -dM -E - | \
+					grep GLIBCXX_RELEASE | sed 's/.*\([1-9][0-9]\)/\1/')
+	if ! [[ ${STDLIBVER} -ge 12 ]]; then
+		die "Hyprland requires >=sys-devel/gcc-12.1.0 to build"
+	fi
```

Comment:
This seems overly complicated when you use *toolchain-funcs* already that provides functions for checking gcc version? Also you might want to move this into `pkg_setup` phase instead.

Reply:
It's overcomplicated, because (when I initially did it), it was possible to build Hyprland with clang, but even when building it with clang, it required GCC-12 to be installed since it linked to libstdc++, but #31084 changes this.

Reply:
But if I understand the correction in #31084 it won't check gcc at all if clang is being used? So it doesn't need gcc anymore?

Reply:
Correct, sorry I wasn't more specific in my previous post.
Newer versions of Hyprland don't link to libstdc++ anymore, when building with clang.